pkg/operator/ceph/csi/csi-spec.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"fmt"
+
+	apps "k8s.io/api/apps/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type CSISpec struct {
+	Namespace string
+
+	RBDPluginImage    string
+	CephFSPluginImage string
+	RegistrarImage    string
+	ProvisionerImage  string
+	AttacherImage     string
+	//TODO: enable snapshot
+
+	CephFSProvisionerSA string
+	CephFSPluginSA      string
+
+	RBDAttacherSA    string
+	RBDProvisionerSA string
+	RBDPluginSA      string
+}
+
+var (
+	CSIParam CSISpec
+
+	EnableRBD    = false
+	EnableCephFS = false
+
+	// image names
+	DefaultRBDPluginImage    = "quay.io/cephcsi/rbdplugin:v1.0.0"
```

Comment:
Can we consider building ceph-csi as a go package into rook? Then we wouldn't need a separate image to run the csi drivers. But we could discuss separate from this PR.

Reply:
yes, that's the eventual goal: the binaries are built as part of rook image. I'll add a comment around it and create an issue for that goal after this.